src/client/utils/searchNarratives.ts

```diff
@@ -84,107 +113,110 @@ export default async function searchNarratives({
-        },
-      });
+      params.access = { only_public: true };
+      params.filters.fields.push({ field: 'is_narratorial', term: true });
       break;
     default:
       throw new Error('Unknown search category');
   }
 
   if (sort) {
     if (sort === 'Recently created') {
-      options.sort = [{ creation_date: { order: 'desc' } }, '_score'];
+      params.sorts = [
+        ['creation_date', SortDir.Desc],
+        ['_score', SortDir.Desc],
+      ];
     } else if (sort === 'Oldest') {
-      options.sort = [{ creation_date: { order: 'asc' } }, '_score'];
+      params.sorts = [
+        ['creation_date', SortDir.Asc],
+        ['_score', SortDir.Desc],
+      ];
     } else if (sort === 'Least recently updated') {
-      options.sort = [{ timestamp: { order: 'asc' } }, '_score'];
+      params.sorts = [
+        ['timestamp', SortDir.Asc],
+        ['_score', SortDir.Desc],
+      ];
     } else if (sort === 'Recently updated') {
-      options.sort = [{ timestamp: { order: 'desc' } }, '_score'];
+      params.sorts = [
+        ['timestamp', SortDir.Desc],
+        ['_score', SortDir.Desc],
+      ];
     } else {
       throw new Error('Unknown sorting method');
     }
   }
-  if (skip) {
-    options.skip = skip;
-  }
-  if (musts.length) {
-    options.query.bool.must = musts;
-  }
-  if (mustNots.length) {
-    options.query.bool.must_not = mustNots;
-  }
-  return await (await makeRequest(options)).result;
+  return await (await makeRequest(params)).result;
 }
 
 /**
  *
- * @param param0 SearchOptions - this takes a query, number of documents to skip,
+ * @param param0 SearchParams - this takes a query, number of documents to skip,
  *  sort parameter, auth (boolean, true if we're looking up personal data), and pageSize
  */
-async function makeRequest({
-  query,
-  skip = 0,
-  sort,
-  auth = true,
-  pageSize,
-}: SearchOptions): Promise<JSONRPCResponse> {
-  const headers = { 'Content-Type': 'application/json' };
-  if (auth) {
+async function makeRequest(params: SearchParams): Promise<JSONRPCResponse> {
+  const headers: { [key: string]: string } = {
+    'Content-Type': 'application/json',
+  };
+  if (!params.access || !params.access.only_public) {
+    // Requires an auth token
     const token = getToken();
     if (!token) {
       throw new Error(
         'Auth token not available for an authenticated search lookup!'
       );
     }
-    Object.assign(headers, { Authorization: token });
-  }
-  const params = {
-    indexes: [INDEX_NAME],
-    size: pageSize,
-    query,
-    from: skip,
-  };
-  if (sort) {
-    let sortObj = sort;
-    Object.assign(params, { sort: sortObj });
+    headers.Authorization = token;
   }
+  console.log('xyz headers', headers);
   const result = await fetch(Runtime.getConfig().service_routes.search, {
     method: 'POST',
     headers,
-    body: JSON.stringify({ method: 'search_objects', params }),
+    body: JSON.stringify({
+      jsonrpc: '2.0',
+      id: Number(new Date()),
```

Comment:
It's possible (but crazy rare) that this might overlap with another user who searches at the same time. Probably better to do Math.random

Reply:
My understanding is that in json-rpc the ID is only for the client, used mainly to figure out the order of the responses after a batch request (and you can leave it out for a notification request). It definitely does not have to be globally unique for the server, or unique across separate client requests.

Reply:
So it would be equally valid to do `id: 0` here

Reply:
The only reason I throw in the timestamp there is that it might benefit the server logs. It doesn't right now. But maybe in the future :shrug:

Reply:
The SDK compiled JSONRPC servers log the ID with the request, theoretically so you can match the request with the logs if the client programmer has an issue. In practice though, the service should generate the ID and pass it to the user rather than relying on all users submitting unique IDs. That's what the auth, groups, and sample services do.

Reply:
Search is not generated with the SDK.. what is the utility of the server generating json-rpc IDs and passing it to the client? 

From the perspective of json-rpc, I'm pretty sure the main functionality of IDs is to order batch responses.

Reply:
Yeah, I'm just providing context for why it's habit for KBase devs to make the id reasonably unique.

Reply:
@MrCreosote That was my understanding, too. But if search isn't recording those ids, and they're only used in the client, then it doesn't matter. I'm fine with it just being the timestamp.